src/services/utils.js

```diff
@@ -0,0 +1,122 @@
+import dayjs from 'dayjs'
+
+const PERIOD = 'DAILY'
+// Percentage for color gradient according to the mean price for the last 7 days
+const PERCENTAGE_MEAN = 10 / 100 // 10%
+
+export function transformIndexedTariffPrices(firstDate, calendarDay, prices) {
+  const first_date = new Date(firstDate)
+  first_date.setHours(0)
+  const calendar_day = new Date(calendarDay)
+  const [startIndex, endIndex] = sliceIndexes(first_date, PERIOD, dayjs(calendar_day))
+
+  var measured_data = timeSlice(first_date, prices, startIndex, endIndex)
+
+  let acum = prices.reduce((accumulator, currentValue) => {
+    return accumulator + currentValue
+  }, 0)
+  let average = acum / prices.length
+
+  const today = new Date()
+  today.setMinutes(0)
+  today.setSeconds(0)
+  today.setMilliseconds(0)
+
+  // build the periods array of dicts
+  let periods = []
+  measured_data.forEach((data) => {
+    const pre = today <= data?.date ? '' : 'past_'
+    // choose the "number" we need
+    if (data.value + average * PERCENTAGE_MEAN < average) {
+      data[`${pre}low`] = data.value
+    } else if (data.value - average * PERCENTAGE_MEAN > average) {
+      data[`${pre}up`] = data.value
+    } else {
+      data[`${pre}average`] = data.value
+    }
+    periods.push(data)
+  })
+  // TODO: check where to define these colors
+  return {
+    fills: {
+      low: '#cbdc49',
+      average: '#71a150',
+      up: '#778462',
+      past_low: '#f0faa5',
+      past_average: '#d6f7be',
+      past_up: '#a5b38f',
+    },
+    keys: ['low', 'average', 'up', 'past_low', 'past_average', 'past_up'],
+    periods: periods,
+  }
+}
+
+// FROM SOM REPRESENTA
+// TODO: move this to somenergia-ui ?
+
+export function time2index(referenceTimestamp, timestamp) {
+  return (new Date(timestamp) - new Date(referenceTimestamp)) / 60 / 60 / 1000
+}
+
+export function index2time(referenceTimestamp, index) {
+  return new Date(new Date(referenceTimestamp).getTime() + index * 60 * 60 * 1000)
+}
+
+export function array2datapoints(first_timestamp, values, step_ms = 60 * 60 * 1000) {
+  const base = first_timestamp.getTime()
+  return values.map((value, i) => {
+    return {
+      date: base + i * step_ms,
+      value,
+    }
+  })
+}
+
+export function timeSlice(timeOffset, values, indexStart, indexEnd) {
+  var adjustedIndexStart = Math.max(0, indexStart)
+  const newTimeOffset = index2time(timeOffset, adjustedIndexStart)
+  return array2datapoints(newTimeOffset, values.slice(adjustedIndexStart, indexEnd))
+}
+
+export function sliceIndexes(offsetDate, period, currentTime) {
+  var [startTime, endTime] = timeInterval(period, currentTime)
+  var startIndex = time2index(offsetDate, startTime)
+  var endIndex = time2index(offsetDate, endTime)
+  return [startIndex, endIndex]
+}
+
+export function timeInterval(scope, current_date) {
+  const start = new Date(current_date)
+  start.setHours(0)
+  start.setMinutes(0)
+  start.setSeconds(0)
+  start.setMilliseconds(0)
+  if (scope === 'MONTHLY') {
```

Comment:
There is a constant defining the scope as DAILY. In this function the scope can varies, is  it because you are thiking in moving this to somenergia-ui?